h2o-py/h2o/estimators/generic.py

```diff
@@ -19,20 +19,29 @@ class H2OGenericEstimator(H2OEstimator):
     """
 
     algo = "generic"
-    param_names = {"model_id", "model_key", "path"}
 
-    def __init__(self, **kwargs):
+    def __init__(self,
+                 model_id=None,  # type: str
+                 model_key=None,  # type: H2OFrame
```

Comment:
the type information doesn't seem right here

Reply:
good catch, for the types, I'm using the type translation that was used for the docstring: there may be some missing conversions: in most cases where we need an H2OFrame, we accept the frame itself or just its key… maybe I should put `Union[str, H2OFrame]` everywhere `H2OFrame` or key is expected. The setter is supposed to extract the key correctly.

Reply:
I see what is going on... it is a bad naming of the parameter. `model_key` is misleading.

The meaning of model_key is key to a binary Vec (wrapped in Frame) that represents the MOJO bytes. So the type should indeed by H2OFrame.

Reply:
OK, here is the distinction:
```
        :param model_key: Key to the self-contained model archive already uploaded to H2O. (default:None).
        :param path: Path to file with self-contained model archive. (default:None).
```
It could (should!) have been a `str` but the client was designed to accept an `H2OFrame` instead for an unknown reason…

Reply:
explanation found in `h2o.upload_mojo`:
```python
    response = api("POST /3/PostFile", filename=mojo_path)
    frame_key = response["destination_frame"]
    mojo_estimator = H2OGenericEstimator(model_key = get_frame(frame_key))
```
I still find this very confusing.

Reply:
Agreed, it is confusing.